plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeBasic.java

```diff
@@ -507,6 +510,46 @@ public void testRequiresQueryPartitionFilterWithUppercaseColumnName()
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    @Test
+    public void testAppendOnly()
+            throws Exception
+    {
+        String tableName = "test_append_only_" + randomNameSuffix();
+        Path tableLocation = Files.createTempFile(tableName, null);
+        copyDirectoryContents(new File(Resources.getResource("deltalake/append_only").toURI()).toPath(), tableLocation);
+        assertUpdate("CALL system.register_table(CURRENT_SCHEMA, '%s', '%s')".formatted(tableName, tableLocation.toUri()));
+        assertQuery("SELECT * FROM " + tableName, "VALUES (1, 11), (2, 12)");
+
+        assertQueryFails("UPDATE " + tableName + " SET a = a + 1", "Cannot modify rows from a table with 'delta.appendOnly' set to true");
+        assertQueryFails("DELETE FROM " + tableName + " WHERE a = 1", "Cannot modify rows from a table with 'delta.appendOnly' set to true");
+        assertQueryFails("DELETE FROM " + tableName, "Cannot modify rows from a table with 'delta.appendOnly' set to true");
+        assertQueryFails("TRUNCATE TABLE " + tableName, "Cannot modify rows from a table with 'delta.appendOnly' set to true");
+        assertQuery("SELECT * FROM " + tableName, "VALUES (1, 11), (2, 12)");
+
+        // Verify delta.appendOnly is preserved after DML
+        assertUpdate("COMMENT ON COLUMN " + tableName + ".a IS 'test column comment'");
+        assertUpdate("COMMENT ON TABLE " + tableName + " IS 'test table comment'");
+        assertUpdate("ALTER TABLE " + tableName + " ADD COLUMN new_col INT");
+        assertThat(query("SELECT * FROM \"" + tableName + "$properties\"")).result().rows()
+                .contains(new MaterializedRow(List.of("delta.appendOnly", "true")));
+    }
+
+    @Test
+    public void testCreateOrReplaceTableOnAppendOnlyTableFails()
+            throws Exception
+    {
+        String tableName = "test_append_only_" + randomNameSuffix();
+        Path tableLocation = Files.createTempFile(tableName, null);
```

Comment:
Or we could even have a method that:

- creates a temp file
- copies given table content to it
- returns a tableLocation (as URI)

Reply:
Maybe this should be `TestTableWithContents(String location)` that is AutoCloseable